api/src/components/link/service.ts

```diff
@@ -124,3 +126,137 @@ export const canLinkBeCreatedBetweenPublicationTypes = (fromType, toType) => {
+                    }
+                };
+            }
+
+            // The link creator is not the owner of the publication
+            if (fromLatestVersion.user.id !== creatorUserId) {
+                return {
+                    valid: false,
+                    details: {
+                        code: 403,
+                        message: `You cannot create a link from the publication with id ${from.publicationId}.`
+                    }
+                };
+            }
+
+            // Peer reviews can only be linked to one thing
+            if (fromPublication.type === 'PEER_REVIEW' && fromPublication.linkedTo.length !== 0) {
+                return {
+                    valid: false,
+                    details: {
+                        code: 403,
+                        message: 'You cannot create another link from a peer review that already has one.'
+                    }
+                };
+            }
+
+            // does a link already exist?
+            const linkExists = await doesLinkExist(from.publicationId, toPublicationId);
+
+            if (linkExists) {
+                return { valid: false, details: { code: 400, message: 'Link already exists.' } };
+            }
+        } else {
+            fromType = from.type;
+        }
+
+        const toPublication = await publicationService.get(toPublicationId);
+
+        if (!toPublication) {
+            return {
+                valid: false,
+                details: { code: 404, message: `Publication to link to with id ${toPublicationId} not found.` }
+            };
+        }
+
+        // Check if publication to be linked to has a live version
+        const toLatestLiveVersion = toPublication.versions.find((version) => version.isLatestLiveVersion);
+
+        if (!toLatestLiveVersion) {
+            return {
+                valid: false,
+                details: {
+                    code: 400,
+                    message: `Publication with id ${toPublicationId} is not LIVE, so a link cannot be created to it.`
+                }
+            };
+        }
+
+        const validPublicationTypes = canLinkBeCreatedBetweenPublicationTypes(fromType, toPublication.type);
+
+        if (!validPublicationTypes) {
+            return {
+                valid: false,
+                details: {
+                    code: 400,
+                    message: `A link cannot be created from a publication of type ${fromType} to a publication of type ${toPublication.type}.`
+                }
+            };
+        }
+
+        return {
+            valid: true,
+            toPublication: {
+                publicationId: toPublication.id,
+                versionId: toLatestLiveVersion.id
+            }
+        };
+    } catch (err) {
+        console.log(err);
+
+        return { valid: false, details: { code: 500, message: 'Unknown server error.' } };
```

Comment:
This might come back to bite you in the future, could you make this message more helpful?

Reply:
I see what you mean, how about something like: 'Link validation failed because of an unexpected error.'?

Reply:
That is better, does the `console.log(err)` go anywhere?

Reply:
It will end up in the lambda logs in cloudwatch, so should be useful to investigate when we see that generic 500 error.

Reply:
:+1: That's what I was looking for. Good stuff!

Reply:
Cool, I have pushed the new error message.